cpp/models/graph_abm/graph_abm_mobility.h

```diff
@@ -0,0 +1,316 @@
+    }
+
+private:
+    std::vector<uint32_t> m_commuting_persons; ///< Person ids that are commuting via an edge
+    std::vector<MobilityRuleType> m_mobility_rules; ///< Rules for moving persons from one node to another
+};
+
+/**
+ * Represents the mobility between two nodes.
+ */
+template <class... History>
+class ABMMobilityEdge
+{
+
+public:
+    using MobilityRuleType = abm::LocationType (*)(const abm::Person&, abm::TimePoint, const abm::Parameters&);
+    /**
+     * Creates edge with mobility parameters
+     * @param params mobility parameters including people commuting via the edge and mobility rules
+     */
+    ABMMobilityEdge(const ABMMobilityParameters& params)
+        : m_parameters(params)
+    {
+    }
+
+    ABMMobilityEdge(const std::vector<uint32_t>& commuting_persons,
+                    const std::vector<MobilityRuleType>& mobility_rules = {})
+        : m_parameters(commuting_persons, mobility_rules)
+    {
+    }
+
+    /**
+     * @brief Get mobility paramters.
+     */
+    const ABMMobilityParameters& get_parameters() const
+    {
+        return m_parameters;
+    }
+
+    /**
+     * @brief Exchanges persons via the edge. 
+     * Commuters are given by the ABMMobilityParameters and exchanged via mobility rules also given ABMMobilityParameters.
+     * @param[in] node_from Commuters home node
+     * @param[in] node_to Node commuters (temporarily) move to
+     * @param[in] t Echange time point
+     */
+    void apply_mobility(ABMSimulationNode<History...>& node_from, ABMSimulationNode<History...>& node_to,
+                        abm::TimePoint t)
+    {
+        // iterate over all persons that could commute via the edge
+        for (auto p : m_parameters.get_commuting_persons()) {
+            auto& person_n1 = node_from.get_simulation().get_world().get_person(mio::abm::PersonId(p));
+            auto& person_n2 = node_to.get_simulation().get_world().get_person(mio::abm::PersonId(p));
+            auto& params    = node_from.get_simulation().get_world().parameters;
+            // as all nodes have all person it doesn't matter which node's persons we take here
+            auto current_location_type = person_n1.get_location_type();
+            auto current_id            = person_n1.get_location();
+            auto current_world_id      = person_n1.get_location_world_id();
+            for (auto& rule : m_parameters.get_mobility_rules()) {
+                auto target_type      = rule(person_n1, t, params);
+                auto target_world_id  = person_n1.get_assigned_location_world_id(target_type);
+                auto target_id        = person_n1.get_assigned_location(target_type);
+                auto& target_world    = (target_world_id == node_from.get_simulation().get_world().get_id())
+                                            ? node_from.get_simulation().get_world()
+                                            : node_to.get_simulation().get_world();
```

Comment:
When the ternary condition is true, then why do we not stop here? The person won't travel along the edge, and just stay in its world - so this should be covered by World::migration, shouldn't it?